File: src/main/java/no/nav/bidrag/beregn/bpsandelunderholdskostnad/BPsAndelUnderholdskostnadCore.java
package no.nav.bidrag.beregn.bpsandelunderholdskostnad;

import no.nav.bidrag.beregn.bpsandelunderholdskostnad.dto.BeregnBPsAndelUnderholdskostnadGrunnlagCore;
import no.nav.bidrag.beregn.bpsandelunderholdskostnad.dto.BeregnetBPsAndelUnderholdskostnadResultatCore;
import no.nav.bidrag.beregn.bpsandelunderholdskostnad.periode.BPsAndelUnderholdskostnadPeriode;

public interface BPsAndelUnderholdskostnadCore {

  BeregnetBPsAndelUnderholdskostnadResultatCore beregnBPsAndelUnderholdskostnad(
      BeregnBPsAndelUnderholdskostnadGrunnlagCore beregnBPsAndelUnderholdskostnadGrunnlagCore);

  static BPsAndelUnderholdskostnadCore getInstance() {
    return new BPsAndelUnderholdskostnadCoreImpl(BPsAndelUnderholdskostnadPeriode.Companion.getInstance());
  }
}

File: src/main/kotlin/no/nav/bidrag/beregn/bpsandelunderholdskostnad/BPsAndelUnderholdskostnadCore.kt
package no.nav.bidrag.beregn.bpsandelunderholdskostnad

import no.nav.bidrag.beregn.bpsandelunderholdskostnad.dto.BeregnBPsAndelUnderholdskostnadGrunnlagCore
import no.nav.bidrag.beregn.bpsandelunderholdskostnad.dto.BeregnetBPsAndelUnderholdskostnadResultatCore
import no.nav.bidrag.beregn.bpsandelunderholdskostnad.periode.BPsAndelUnderholdskostnadPeriode

fun interface BPsAndelUnderholdskostnadCore {
    fun beregnBPsAndelUnderholdskostnad(grunnlag: BeregnBPsAndelUnderholdskostnadGrunnlagCore): BeregnetBPsAndelUnderholdskostnadResultatCore

    companion object {
        fun getInstance(): BPsAndelUnderholdskostnadCore = BPsAndelUnderholdskostnadCoreImpl(BPsAndelUnderholdskostnadPeriode.getInstance())
    }
}
